src/main/java/frc/robot/Robot.java

```diff
@@ -93,7 +95,7 @@ public void robotInit() {
     this.hopperController = new HopperController();
```

Comment:
You need to Construct the distanceSensorDetector before you can use it.
add this.distanceSensorDetector = new DistanceSensorDetector(); before you create Operator Inputs